modules/js/perf/perf_imgproc/perf_sobel.js

```diff
@@ -25,43 +25,40 @@ function perf() {
+    const Sobel3x3dxdy = ["(0,1)", "(1,0)", "(1,1)", "(0,2)", "(2,0)", "(2,2)"];
+    const Sobeldxdy = ["(0,1)", "(1,0)", "(1,1)", "(0,2)", "(2,0)"];
+    const BorderType3x3 = ["BORDER_REPLICATE", "BORDER_CONSTANT"];
+    const BorderType3x3ROI = ["BORDER_DEFAULT", "BORDER_REPLICATE|BORDER_ISOLATED", "BORDER_CONSTANT|BORDER_ISOLATED"];
+    const BorderType = ["BORDER_REPLICATE", "BORDER_CONSTANT", "BORDER_REFLECT", "BORDER_REFLECT101"];
+    const BorderTypeROI = ["BORDER_DEFAULT", "BORDER_REPLICATE|BORDER_ISOLATED", "BORDER_CONSTANT|BORDER_ISOLATED", "BORDER_REFLECT|BORDER_ISOLATED", "BORDER_REFLECT101|BORDER_ISOLATED"]
 
-    const combiSobelMode8U = combine(SobelSize, ["CV_8U"], ["CV_8U", "CV_16S", "CV_32F", "CV_64F"], SobelDirection, SobelKsize, SobelBoderType);
-    const combiSobelMode16U = combine(SobelSize, ["CV_16U"], ["CV_16U", "CV_32F", "CV_64F"], SobelDirection, SobelKsize, SobelBoderType);
-    const combiSobelMode16S = combine(SobelSize, ["CV_16S"], ["CV_16S", "CV_32F", "CV_64F"], SobelDirection, SobelKsize, SobelBoderType);
-    const combiSobelMode32F = combine(SobelSize, ["CV_32F"], ["CV_32F", "CV_64F"], SobelDirection, SobelKsize, SobelBoderType);
-    const combiSobelMode64F = combine(SobelSize, ["CV_64F"], ["CV_64F"], SobelDirection, SobelKsize, SobelBoderType);
+    const combiSobelBorder3x3 = combine(SobelSize, ["CV_16SC1", "CV_32FC1"], Sobel3x3dxdy, BorderType3x3);
+    const combiSobelBorder3x3ROI = combine(SobelSize, ["CV_16SC1", "CV_32FC1"], Sobel3x3dxdy, BorderType3x3ROI);
+    const combiSobelBorder5x5 = combine(SobelSize, ["CV_16SC1", "CV_32FC1"], Sobeldxdy, BorderType);
+    const combiSobelBorder5x5ROI = combine(SobelSize, ["CV_16SC1", "CV_32FC1"], Sobeldxdy, BorderTypeROI);
 
     function addSobelCase(suite, type) {
         suite.add('sobel', function() {
             cv.Sobel(src, dst, ddepth, dx, dy, ksize, 1, 0, borderType);
           }, {
               'setup': function() {
                 let size = this.params.size;
-                let matType = cv[this.params.matType];
                 let ddepth = cv[this.params.ddepth];
-                let direction = this.params.direction;
-                let ksize = parseInt(this.params.ksize.split("_").pop());
-                let borderType =cv[this.params.borderType];
-                let src = new cv.Mat(size, matType);
-                let dst = new cv.Mat(size, matType);
-                let dx = 0;
-                let dy = 0;
-
-                if (ddepth == matType) {
-                  ddepth = -1;
-                }
+                let dxdy = this.params.dxdy;
+                let ksize = this.params.ksize;
                 
-                if (direction == "dx") {dx = 1;}
-                else {dy = 1;}
+                let src = new cv.Mat(size[1], size[0], cv.CV_8U);
+                let dst = new cv.Mat(size[1], size[0], ddepth);
+                let dx = parseInt(dxdy[1]);
+                let dy = parseInt(dxdy[dxdy.length-2]);
```

Comment:
Is evaluation `dxdy.length-2` really necessary? I suppose the result is always equal to zero.

Reply:
I think it is necessary. The parameter `dxdy` represents the tuple of the `dx` and `dy`.  In the native perf test,  there are some options of the tuple of `dx` and `dy`, like `(0, 1)`, `(1, 0)`, `(1, 1)`, `(2, 0)` and so on.

Reply:
Looks like the tuple length is constant and always equals 2. Anyway if the tuple length could variate do we always want second from the end value?

Reply:
I think we need the second value. If we just get one value `1` only, we do not understand if the user wants to set the `dx` as 1 or set `dy` as 1. And if we get one value `2` only, we do not understand if the user wants to set the `dx` as 2 or set `dy` as 2, or set `dx` and `dy` as 1 together, where their sum is 2.

Reply:
I get the idea. Sorry I've messed the length of the string with amount of the tuple elements. My bad.

Anyway 3 should work better for persons not familiar with javascript like me.